flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1338,14 +1368,23 @@ private boolean restoreLatestCheckpointedStateInternal(
 			CompletedCheckpoint latest = completedCheckpointStore.getLatestCheckpoint(isPreferCheckpointForRecovery);
 
 			if (latest == null) {
+				LOG.info("No checkpoint found during restore.");
+
 				if (errorIfNoCheckpoint) {
 					throw new IllegalStateException("No completed checkpoint available");
-				} else {
+				}
+
+				if (coordinatorRestoreBehavior == CoordinatorRestore.ALWAYS) {
```

Comment:
nit: how about `if (coordinatorRestoreBehavior.resetIfNoState())` ?

Reply:
If this becomes `if (coordinatorRestoreBehavior == RESTORE_OR_RESET)` then this looks good, I think.
It is also more about "no checkpoint" than "no state". "No state" can also be the case if there is a checkpoint.